src/autolabel/schema.py

```diff
@@ -209,6 +210,24 @@ class GenerationCacheEntry(BaseModel):
     class Config:
         orm_mode = True
 
+    def get_id(self) -> str:
+        """
+        Generates a unique ID for the given generation cache configuration
+        """
+        return calculate_md5([self.model_name, self.model_params, self.prompt])
+
+    def get_serialized_output(self) -> str:
+        """
+        Returns the serialized cache entry output
+        """
+        return json.dumps([gen.dict() for gen in self.generations])
```

Comment:
For things like bytes objects which come as a result of transforms, json dumping fails. That is why we switched to serializing using pickle.

Reply:
Makes sense, I can move to pickle in a follow up PR